### ISSUE TITLE
Benchdnn int8 verification value

### ISSUE THREAD

**Issue by qt-suzukik** [oneapi-src/oneDNN#892]
As input data for U8 in benchdnn, the value is limited to between 0 and 8.

https://github.com/oneapi-src/oneDNN/blob/master/tests/benchdnn/conv/cfg.cpp#L138
```cpp
const _dt_conf_t conf_u8s8f32 = {
        {dnnl_u8, 0, UINT8_MAX, 0, 8, 0, 1, .25, 0.},
        {dnnl_s8, INT8_MIN, INT8_MAX, -5, 5, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -8, 32, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -255, 255, 0, 1, .25, 0.},
        {dnnl_s32},
};
```

Wouldn't it be appropriate to fix this as follows?

```diff
const _dt_conf_t conf_u8s8f32 = {
-        {dnnl_u8, 0, UINT8_MAX, 0, 8, 0, 1, .25, 0.},
+        {dnnl_u8, 0, UINT8_MAX, 0, UINT8_MAX, 0, 1, .25, 0.},
        {dnnl_s8, INT8_MIN, INT8_MAX, -5, 5, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -8, 32, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -255, 255, 0, 1, .25, 0.},
        {dnnl_s32},
};
```

Our JIT implementation for AArch64, which benchdnn has passed, has been tried with real data in conjunction with TensorFlow, and has had problems with training accuracy.

I was using s8 instructions when I should have been using AArch64's u8 instructions.
It was the cause, but it was not detected by benchdnn.

When fixed as described above, it was FAIL in some patterns and correctly detected the bug.

**Comment by dzarukin** [comment]
Hi @qt-suzukik, thanks for your question.

Benchdnn uses [0-8] range for u8 to avoid overflow (and further saturation) just with a single src[i] * wei[i] result in the output. Such change would likely make most int8 convolution outputs contain either 0 or UINT8_MAX values which won't help for proper validation.

Do I get the problem right that due to benchdnn fills with values from common to u8 and s8 range, it doesn't make any difference how to load the data, whether with s8 or u8 instruction, but in fact, if the data was from mutually exclusive ranges, than the error would pop up?

Thanks.

**Comment by t-hasimo** [comment]
Dear @dzarukin,

While the operands of the 8-bit integer dot4 product instruction (VPDPBUSD) of x86_64 are src1_dtype=u8, src2_dtype=s8 and dst_dtype=s32, the operands of the 8-bit integer dot4 product instruction (SDOT) of aarch64 are src1_dtype=s8, src2_dtype=s8 and dst_dtype=s32. 

If the u8 source data is in the range from 0 to 127, SDOT of aarch64 will not make any difference. But, if the u8 source data is in the range from 128 to 255, then s8 is [-128,-1] and the computation results of SDOT and VPDPBUSD will not match.

Then, in the case of u8s8s8, u8s8u8, u8s8s32 and u8s8f32, the int8 convolution using SDOT of aarch64 comuputes as:
```
dst_s32 <- 2 * (s32) ((src_u8 - 128) (*) (wei_s8 / 2)) + 128 (*) wei_s8
```

Since benchdnn does not include [128-255] in the input test data, it is unable to validate the above function. For this reason, we would like you to add a pattern including [128-255] to u8s8s8, u8s8u8, u8s8s32 and u8s8f32 as follows:
```
const _dt_conf_t conf_u8s8f32 = {
        {dnnl_u8, 0, UINT8_MAX, 0, 8, 0, 1, .25, 0.},
+        {dnnl_u8, 0, UINT8_MAX, 0, UINT8_MAX, 0, 1, .25, 0.},
        {dnnl_s8, INT8_MIN, INT8_MAX, -5, 5, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -8, 32, 0, 1, .25, 0.},
        {dnnl_f32, INT32_MIN, INT32_MAX, -255, 255, 0, 1, .25, 0.},
        {dnnl_s32},
};
```

**Comment by dzarukin** [comment]
Hi @t-hasimo, thanks for the explanation, I see it now. In my first answer I wrote it will bring more MISTRUSTED results having in mind u8s8**s8** or u8s8**u8** combinations. But for **s32** or **f32** dst data type it may actually work. Let me make a patch, test it internally and come back with finding whether it's a good change. Thanks.

**Comment by dzarukin** [comment]
Hi @t-hasimo, @qt-suzukik.
The testing proved the change is valid, passed the correctness checks and didn't bring any instabilities. It will be promoted to master soon. Thank you for the idea and making oneDNN testing better.

P.S. It's interesting that we have never stepped on such issue before (or at least no one raised a concern about it).

**Comment by qt-suzukik** [comment]
Hi @dzarukin thank you for testing.
I'm glad to contribute to the testing of oneDNN.
When should this Issue be closed? As for me, it's okay to close now.

@t-hasimo thank you for following.

Thanks.

**Comment by vpirogov** [comment]
Fixed by 1ae8a6107c13bc77ef254e00aa27dbf95603376c.
